src/JwtFacade.php

```diff
@@ -0,0 +1,45 @@
+<?php
+declare(strict_types=1);
+
+namespace Lcobucci\JWT;
+
+use Lcobucci\JWT\Encoding\ChainedFormatter;
+use Lcobucci\JWT\Encoding\JoseEncoder;
+use Lcobucci\JWT\Token\Parser;
+use Lcobucci\JWT\Validation\Constraint;
+use Lcobucci\JWT\Validation\SignedWith;
+use Lcobucci\JWT\Validation\ValidAt;
+use Lcobucci\JWT\Validation\Validator;
+
+use function assert;
+
+final class JwtFacade
+{
+    public function getBuilder(): Builder
```

Comment:
Instead of https://github.com/lcobucci/jwt/pull/759#discussion_r649256312 I've chosen to only have this as token creation help because:

1. If we accept `array<string, mixed> $claims` than we have to map them back to `Builder` interface, which would just be a nightmare
2. `Builder` is already a rock solid interface, easy to deal with within IDEs and cannot be misused. I'll implement soon a `ExpirationRequiredBuilder` just to be sure users calls `expiresAt`
3. It would also result in a one-liner code, just like the test shows

Reply:
For point [2] have a look at the new commit https://github.com/lcobucci/jwt/pull/759/commits/f539fcc2f63a25fee42b9e059eda0070be34504d

Reply:
I'd prefer us containing the impact to only this component.
Instead of using an array of claims, we could use a closure:

```php
/**
 * @param Closure(Builder):Builder $customiseBuilder
 */
public function issueToken(Signer $signer, Key $signingKey, Closure $customiseBuilder): UnencryptedToken
{
    $builder = new Token\Builder(new JoseEncoder(), ChainedFormatter::withUnixTimestampDates());
    $now = new DateTimeImmutable();

    return $customiseBuilder($builder)
        ->issuedAt($now)
        ->canOnlyBeUsedAfter($now)
        ->expiresAt($now->modify('+5 minutes'))
        ->getToken($signer, $signingKey);
}
```

We can surely modify the position of the date claims configuration but I think this combines the best of both worlds. We make sure the time constraints are set and users can rely on the `Builder` interface to perform the customisation (without having to maintain other components).

Reply:
I've set timestamps before calling the closure, so the wise user can overwrite them - but never unset them :wink: